### PR TITLE
chore: test node TLS and Wallet connections

### PR DESCRIPTION
**Pull request by sshaman1101**

### CODE REVIEW

File: insonmnia/node/server.go [hunk 1]
```diff
@@ -293,6 +294,7 @@ func (n *Node) ServeGRPC() error {
 		if err == nil {
 			log.S(n.ctx).Infof("starting node %s listener on %s", netFam, lis.Addr().String())
 			wg.Go(func() error {
+				n.listeners = append(n.listeners, lis)
```
Comment (noxiouz) [inline insonmnia/node/server.go]: Looks like possible data race, as n.listeners are modified from two goroutines

File: insonmnia/node/server_test.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func newTestNode(t *testing.T, key *ecdsa.PrivateKey) *Node {
+	relayAddr := netutil.TCPAddr{}
+	relayAddr.TCPAddr = net.TCPAddr{IP: net.IPv4(127, 0, 0, 1), Port: 12345}
+
+	ctx := context.Background()
+	nod, err := New(ctx, &Config{
```
Comment (3Hren) [inline insonmnia/node/server_test.go]: image

File: insonmnia/node/server_test.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+	return nod
+}
+
+func TestConnectWithoutTLS(t *testing.T) {
+	key, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	nod := newTestNode(t, key)
+	go nod.Serve()
```
Comment (3Hren) [inline insonmnia/node/server_test.go]: image

File: insonmnia/node/server_test.go [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+	assert.True(t, ok, "error must be provided by GRPC")
+	assert.Equal(t, codes.Unavailable, grpcErr.Code(), "must be unable to connect")
+}
+
+func TestConnectWithValidKeyWithoutWallet(t *testing.T) {
+	nodeKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	nod := newTestNode(t, nodeKey)
+	go nod.Serve()
```
Comment (3Hren) [inline insonmnia/node/server_test.go]: image

File: insonmnia/node/server_test.go [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+	_, err = sonm.NewMarketClient(cc).GetOrderByID(ctx, &sonm.ID{Id: "1"})
+	require.NoError(t, err)
+}
+
+func TestConnectWithInvalidKeyWithoutWallet(t *testing.T) {
+	nodeKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	nod := newTestNode(t, nodeKey)
+	go nod.Serve()
```
Comment (3Hren) [inline insonmnia/node/server_test.go]: image

File: insonmnia/node/server_test.go [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+	assert.True(t, ok, "error must be provided by GRPC")
+	assert.Equal(t, codes.Unavailable, grpcErr.Code(), "must be unable to connect")
+}
+
+func TestConnectWithInvalidKeyWithWallet(t *testing.T) {
+	nodeKey, err := crypto.GenerateKey()
+	require.NoError(t, err)
+
+	nod := newTestNode(t, nodeKey)
+	go nod.Serve()
```
Comment (3Hren) [inline insonmnia/node/server_test.go]: image

**Pull request merged**